lightning/src/routing/scoring.rs

```diff
@@ -259,6 +259,65 @@ impl ReadableArgs<u64> for FixedPenaltyScorer {
 	}
 }
 
+/// [`Score`] implementation that uses a dynamic penalty.
+///
+/// (C-not exported) as we don't implement function bounds mapping yet.
+pub struct DynamicPenaltyScorer<F>
+where
+	F: Fn(u64, &NodeId, &NodeId, ChannelUsage) -> u64,
```

Comment:
Not necessary anymore, since https://github.com/lightningdevkit/rust-lightning/pull/1491/commits/19ecb4c02798caf3d89fd42e9497c78565efe756 got rid of `DynamicPenaltyScorer`.